File: Python/7py 1/explode.py
"""
|--------------------------------------------------------------------------
| Digits explosion
|--------------------------------------------------------------------------
|
"""

def explode(s):
   res = ''
   for i in range(len(s)):
      for j in range(int(s[i])):
         res += s[i]
   return res


def explode(s):
   return ''.join(c * int(c) for c in s)


def explode(s):
   result = ""
   for num in s:
      result += int(num)*num
   return result


def explode(string: str) -> str:
   return "".join(map(lambda _it: _it * int(_it), string))


print(explode("312"), "333122")
print(explode("102269"),"12222666666999999999")
print(explode("0"), "")
print(explode("000"), "")
print(explode("123"), "122333")

File: JavaScript/7js 1/explode.js
/*
|--------------------------------------------------------------------------
| Digits explosion
|--------------------------------------------------------------------------
|
*/

function explode(s) {
   let res = '';
   for (let i = 0; i < s.length; i++) {
      for (let j = 0; j < Number(s[i]); j++) {
         res += s[i];
      }
   }
   return res;
}


const explode = s => s.replace(/\d/g, d => d.repeat(d));


function explode(s) {
   return s.split("").map((e) => e.repeat(+e)).join("");
}


function explode(s) {
   return s.split('').map(i => ''.padEnd(i, i)).join('')
}


console.log(explode("312"));
console.log(explode("102269"));